src/Microsoft.IdentityModel.Tokens/CryptoProviderFactory.cs

```diff
@@ -34,13 +34,13 @@ namespace Microsoft.IdentityModel.Tokens
     /// Creates <see cref="SignatureProvider"/>s by specifying a <see cref="SecurityKey"/> and algorithm.
     /// <para>Supports both <see cref="AsymmetricSecurityKey"/> and <see cref="SymmetricSecurityKey"/>.</para>
     /// </summary>
-    public class SignatureProviderFactory
+    public class CryptoProviderFactory
```

Comment:
@PinpointTownes @tushargupta5 My feeling is that as we add new crypto, we would need mods to each key, however I am will to consider this given
>  BTW. For instance, there's no encrypting equivalent for ECDsa
I would like to get this PR in as we need to get the JWE prototype going which needs this PR. Can or did we open an issue here?